src/snkit/network.py

```diff
@@ -315,28 +315,15 @@ def split_edges_at_intersections(network, tolerance=1e-9):
             # these are *new geometrical objects* not in the sindex
             # therefore they cannot be returned by internal _intersects*
             intersection = Point()
-            # restrict intersection to crossing
-            # this excludes self overlap and end-points intersections
-            # WARNING: how are self-crossing handled? (e.g. loop-like)
-            if edge.geometry.crosses(hit):
+            # excluding itself
+            # note that __eq__ is used on purpose instead of equals()
+            # this is stricter: for geometries constructed in the same way
+            # WARNING: this excludes sensical self-intersections, to be solved.
+            if edge != hit:
                 intersection = edge.geometry.intersection(hit)
```

Comment:
Perhaps the else case should be:
```python
            else:
                intersection = shapely.ops.unary_union(edge.geometry)
```

Reply:
Yes I'm indeed working with `unary_union` for the "self-split". However I have other issues. I'll elaborate above.